packages/ckeditor5-list/src/liststylescommand.js

```diff
@@ -0,0 +1,206 @@
+			...getSiblingNodes( position, 'backward' ),
+			...getSiblingNodes( position, 'forward' )
+		];
+
+		model.change( writer => {
+			for ( const item of listItems ) {
+				writer.setAttribute( 'listStyle', options.type || 'default', item );
+			}
+		} );
+	}
+
+	/**
+	 * Checks the command's {@link #value}.
+	 *
+	 * @private
+	 * @returns {String|null} The current value.
+	 */
+	_getValue() {
+		const listItem = this.editor.model.document.selection.getFirstPosition().parent;
+
+		if ( listItem && listItem.is( 'element', 'listItem' ) ) {
+			return listItem.getAttribute( 'listStyle' );
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks whether the command can be enabled in the current context.
+	 *
+	 * @private
+	 * @returns {Boolean} Whether the command should be enabled.
+	 */
+	_checkEnabled() {
+		const editor = this.editor;
+
+		const numberedList = editor.commands.get( 'numberedList' );
+		const bulletedList = editor.commands.get( 'bulletedList' );
+
+		return numberedList.isEnabled || bulletedList.isEnabled;
+	}
+}
+
+// Returns a position that is hooked in the `listItem` element.
+// It can be the selection starting position or end.
+//
+// @param {module:engine/model/selection~Selection} selection
+// @returns {module:engine/model/position~Position|null}
+function findPositionInsideList( selection ) {
+	const startPosition = selection.getFirstPosition();
+
+	if ( startPosition.parent.is( 'element', 'listItem' ) ) {
+		return startPosition;
+	}
+
+	const lastPosition = selection.getLastPosition();
+
+	if ( lastPosition.parent.is( 'element', 'listItem' ) ) {
+		return lastPosition;
+	}
+
+	return null;
+}
+
+// Returns an array with all `listItem` elements that represents the same list.
+//
+// It means that values for `listIndent`, `listType`, and `listStyle` for all items
+// are equal.
+//
+// @param {module:engine/model/position~Position} position Starting position.
+// @param {'forward'|'backward'} direction Walking direction.
+// @returns {Array.<module:engine/model/element~Element>
+function getSiblingNodes( position, direction ) {
+	const items = [];
+	const listItem = position.parent;
+	const walkerOptions = {
+		ignoreElementEnd: true,
+		startPosition: position,
+		shallow: true,
+		direction
+	};
+	const limitIndent = listItem.getAttribute( 'listIndent' );
+	const nodes = [ ...new TreeWalker( walkerOptions ) ]
+		.filter( value => value.item.is( 'element' ) )
+		.map( value => value.item );
+
+	for ( const element of nodes ) {
+		// If found something else than `listItem`, we're out of the list scope.
+		if ( !element.is( 'element', 'listItem' ) ) {
+			break;
+		}
+
+		// If current parsed item has lower indent that element that the element that was a starting point,
+		// it means we left a nested list. Abort searching items.
+		//
+		// ■ List item 1.       [listIndent=0]
+		//     ○ List item 2.[] [listIndent=1], limitIndent = 1,
+		//     ○ List item 3.   [listIndent=1]
+		// ■ List item 4.       [listIndent=0]
```

Comment:
:+1:  for styling :)

Reply:
Thx :D I was wondering how to explain what I am doing in particular functions. I guess this "trick" explains everything.